### PR TITLE
WIP: using the invented wheels

### PR DESCRIPTION
**Pull request by hypeJunction**
This starts the process of cleaning up the lists and using existing list components, subsequent to #7903

### CODE REVIEW

**Comment by jdalsem** [comment]
wow really love the cleanup work. Good to see LOC reduction and improved readability

**Comment by ewinslow** [comment]
Yay cleanup!

**Comment by hypeJunction** [comment]
There will be lots to clean up if we get #7903 in

**Comment by hypeJunction** [comment]
Views that reinvent the wheel
- [ ] mod\groups\views\default\groups\membershiprequests.php
- [ ] mod\search\views\default\search\list.php
- [ ] mod\uservalidationbyemail\views\default\forms\uservalidationbyemail\bulk_action.php

File: mod/groups/views/default/group/format/invitationrequest.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+$user = elgg_get_logged_in_user_entity();
+$group = elgg_extract('entity', $vars);
+
+if (!elgg_instanceof($group, 'group')) {
+	return true;
+}
+
+$icon = elgg_view_entity_icon($group, 'small');
+
+$url = elgg_add_action_tokens_to_url(elgg_get_site_url() . "action/groups/join?user_guid={$user->guid}&group_guid={$group->guid}");
```
Comment (jdalsem) [inline mod/groups/views/default/group/format/invitationrequest.php]: why not just use `is_action => true` in `output/url`

File: mod/groups/views/default/group/format/invitationrequest.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+$accept_button = elgg_view('output/url', array(
+	'href' => $url,
+	'text' => elgg_echo('accept'),
+	'class' => 'elgg-button elgg-button-submit',
+	'is_trusted' => true,
+		));
+
+$url = "action/groups/killinvitation?user_guid={$user->getGUID()}&group_guid={$group->getGUID()}";
```
Comment (jdalsem) [inline mod/groups/views/default/group/format/invitationrequest.php]: doesn't this need action tokens? `is_action => true`

**Comment by hypeJunction** [comment]
Will collapse these into multiple PRs

**Pull request closed without merging**